core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/Retry.scala

```diff
@@ -33,16 +37,27 @@ trait Retry {
    * @tparam T
    * @return
    */
-  def retryUntilSuccessOrTimeout[T](pred: (Throwable) => Boolean = always)(interval: FiniteDuration, timeout: FiniteDuration)(op: () => Future[T])(implicit executionContext: ExecutionContext): Future[T] = {
+  def retryUntilSuccessOrTimeout[T](pred: (Throwable) => Boolean = always, failureLogMessage: String = defaultErrorMessage)(interval: FiniteDuration, timeout: FiniteDuration)(op: () => Future[T])(implicit executionContext: ExecutionContext): Future[T] = {
     val trialCount = Math.ceil(timeout / interval).toInt
-    retry(Seq.fill(trialCount)(interval))(op,pred)
+    retry(Seq.fill(trialCount)(interval))(op, pred, failureLogMessage)
   }
 
-  private def retry[T](remainingBackoffIntervals: Seq[FiniteDuration])(op: => () => Future[T], pred: (Throwable) => Boolean)(implicit executionContext: ExecutionContext): Future[T] = {
+  private def retry[T](remainingBackoffIntervals: Seq[FiniteDuration])(op: => () => Future[T], pred: (Throwable) => Boolean, failureLogMessage: String)(implicit executionContext: ExecutionContext): Future[T] = {
     op().recoverWith {
-      case t if pred(t) && !remainingBackoffIntervals.isEmpty => after(remainingBackoffIntervals.head, system.scheduler) {
-        retry(remainingBackoffIntervals.tail)(op, pred)
-      }
+      case t if pred(t) && !remainingBackoffIntervals.isEmpty =>
+        logger.info(s"$failureLogMessage: ${remainingBackoffIntervals.size} retries remaining, retrying in ${remainingBackoffIntervals.head}", t)
+        after(remainingBackoffIntervals.head, system.scheduler) {
+          retry(remainingBackoffIntervals.tail)(op, pred, failureLogMessage)
+        }
+
+      case t =>
+        if (remainingBackoffIntervals.isEmpty) {
+          logger.info(s"$failureLogMessage: no retries remaining", t)
+        } else {
+          logger.info(s"$failureLogMessage: retries remain but predicate failed, not retrying", t)
```

Comment:
It would be nice to know what the predicate is here, but it doesn't vary too much from what I can tell. At least we have `t` logged which is probably the more important piece.